From 091274470bd8bac7de06daf230edc753a2d25376 Mon Sep 17 00:00:00 2001
From: Randy Li <randy.li@rock-chips.com>
Date: Mon, 17 Apr 2017 10:22:13 +0800
Subject: [PATCH] video: rockchip: vpu: add avs decoder table

This table would be used for both AVS and AVS+ decoder.

Change-Id: I9557a3d170943a3b544d97b6c63f02679bd7b532
Signed-off-by: Randy Li <randy.li@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_hw_vpu.h | 30 +++++++++++--------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_hw_vpu.h b/drivers/video/rockchip/vcodec/vcodec_hw_vpu.h
index 6f0a3005b45b..ae771b62a7c4 100644
--- a/drivers/video/rockchip/vcodec/vcodec_hw_vpu.h
+++ b/drivers/video/rockchip/vcodec/vcodec_hw_vpu.h
@@ -277,6 +277,10 @@ DEF_FMT_TRANS_TBL(vpu_vc1d,
 		  12, 13, 14, 15, 16, 17, 27, 41
 );
 
+DEF_FMT_TRANS_TBL(vpu_avsd,
+		  12, 13, 14, 15, 16, 17, 40, 41, 45
+);
+
 DEF_FMT_TRANS_TBL(vpu_defaultd,
 		  12, 13, 14, 15, 16, 17, 40, 41
 );
@@ -293,29 +297,29 @@ DEF_FMT_TRANS_TBL(vpu_defaulte,
 		  5, 6, 7, 8, 9, 10, 11, 12, 13, 51
 );
 
-const struct vpu_trans_info trans_vpu[FMT_TYPE_BUTT] = {
-	SETUP_FMT_TBL(FMT_JPEGD , vpu_jpegd),
-	SETUP_FMT_TBL(FMT_H263D , vpu_defaultd),
-	SETUP_FMT_TBL(FMT_H264D , vpu_h264d),
+static const struct vpu_trans_info trans_vpu[FMT_TYPE_BUTT] = {
+	SETUP_FMT_TBL(FMT_JPEGD, vpu_jpegd),
+	SETUP_FMT_TBL(FMT_H263D, vpu_defaultd),
+	SETUP_FMT_TBL(FMT_H264D, vpu_h264d),
 	EMPTY_FMT_TBL(FMT_H265D),
 
 	SETUP_FMT_TBL(FMT_MPEG1D, vpu_defaultd),
 	SETUP_FMT_TBL(FMT_MPEG2D, vpu_defaultd),
 	SETUP_FMT_TBL(FMT_MPEG4D, vpu_defaultd),
 
-	SETUP_FMT_TBL(FMT_VP6D  , vpu_vp6d),
-	SETUP_FMT_TBL(FMT_VP7D  , vpu_defaultd),
-	SETUP_FMT_TBL(FMT_VP8D  , vpu_vp8d),
+	SETUP_FMT_TBL(FMT_VP6D, vpu_vp6d),
+	SETUP_FMT_TBL(FMT_VP7D, vpu_defaultd),
+	SETUP_FMT_TBL(FMT_VP8D, vpu_vp8d),
 	EMPTY_FMT_TBL(FMT_VP9D),
 
-	SETUP_FMT_TBL(FMT_VC1D  , vpu_vc1d),
-	SETUP_FMT_TBL(FMT_AVSD  , vpu_defaultd),
+	SETUP_FMT_TBL(FMT_VC1D, vpu_vc1d),
+	SETUP_FMT_TBL(FMT_AVSD, vpu_avsd),
 
-	SETUP_FMT_TBL(FMT_PP    , vpu_default_pp),
+	SETUP_FMT_TBL(FMT_PP, vpu_default_pp),
 
-	SETUP_FMT_TBL(FMT_JPEGE , vpu_defaulte),
-	SETUP_FMT_TBL(FMT_H264E , vpu_defaulte),
-	SETUP_FMT_TBL(FMT_VP8E  , vpu_vp8e),
+	SETUP_FMT_TBL(FMT_JPEGE, vpu_defaulte),
+	SETUP_FMT_TBL(FMT_H264E, vpu_defaulte),
+	SETUP_FMT_TBL(FMT_VP8E, vpu_vp8e),
 };
 
 #endif
-- 
2.35.3

